haystack/document_store/faiss.py

```diff
@@ -122,34 +132,63 @@ def update_embeddings(self, retriever: BaseRetriever, index: Optional[str] = Non
         for i, doc in enumerate(documents):
             doc.embedding = embeddings[i]
 
-        phi = self._get_phi(documents)
-
         vector_id_map = {}
         for i in range(0, len(documents), self.index_buffer_size):
             vector_id = self.faiss_index.ntotal
             embeddings = [doc.embedding for doc in documents[i: i + self.index_buffer_size]]
-            hnsw_vectors = self._get_hnsw_vectors(embeddings=embeddings, phi=phi)
-            hnsw_vectors = hnsw_vectors.astype(np.float32)
-            self.faiss_index.add(hnsw_vectors)
+            embeddings = np.array(embeddings, dtype="float32")
+            self.faiss_index.add(embeddings)
 
             for doc in documents[i: i + self.index_buffer_size]:
                 vector_id_map[doc.id] = vector_id
                 vector_id += 1
                 
         self.update_vector_ids(vector_id_map, index=index)
 
+    def train_index(self, documents: Optional[Union[List[dict], List[Document]]], embeddings: Optional[np.array] = None):
+        """
+        Some FAISS indices (e.g. IVF) require initial "training" on a sample of vectors before you can add your final vectors.
+        The train vectors should come from the same distribution as your final ones.
+        You can pass either documents (incl. embeddings) or just the plain embeddings that the index shall be trained on.
+
+        :param documents: Documents (incl. the embeddings)
+        :param embeddings: Plain embeddings
+        :return: None
+        """
+
+        if embeddings and documents:
+            raise ValueError("Either pass `documents` or `embeddings`. You passed both.")
+        if documents:
+            document_objects = [Document.from_dict(d) if isinstance(d, dict) else d for d in documents]
+            embeddings = [doc.embedding for doc in document_objects]
+            embeddings = np.array(embeddings, dtype="float32")
+        self.faiss_index.train(embeddings)
```

Comment:
Need to check `self.faiss_index.is_trained` before calling train otherwise function will fail.

Reply:
Ok, we could catch the error and make it more descriptive, but I think we should also let the user know that it's not valid to call train_index() in this case.